### PR TITLE
feat: embed version info into binary

### PR DESCRIPTION
**Pull request by johnstcn**
Adds version info to envbuilder binary. Most of the logic is fork-lifted from `coder/coder`.

### CODE REVIEW

File: scripts/version.sh [hunk 1]
```diff
@@ -1,10 +1,77 @@
...
+fi
+
+# To make contributing easier, if there are no tags, we'll use a default
+# version.
+tag_list=$(git tag)
+if [[ -z ${tag_list} ]]; then
+	log
+	log "INFO(version.sh): It appears you've checked out a fork or shallow clone of Envbuilder."
+	log "INFO(version.sh): By default GitHub does not include tags when forking."
+	log "INFO(version.sh): We will use the default version 2.0.0 for this build."
```
Comment (dannykopping) [inline scripts/version.sh:30]: Is v2.0.0 legit?

**Review by dannykopping — APPROVED** [review]

File: scripts/version.sh [hunk 2]
```diff
@@ -1,10 +1,77 @@
...
+	log "INFO(version.sh): To pull tags from upstream, use the following commands:"
+	log "INFO(version.sh):   - git remote add upstream https://github.com/coder/envbuilder.git"
+	log "INFO(version.sh):   - git fetch upstream"
+	log
+	last_tag="v2.0.0"
```
Comment (mafredri) [inline scripts/version.sh:35]: Fixme?

File: scripts/version.sh (hunk 1 above)
Comment (mafredri) [inline scripts/version.sh:30]: Do we use 0.0.1 for envbuilder, perhaps?

File: scripts/version.sh [hunk 3]
```diff
@@ -1,10 +1,77 @@
...
+	if [[ -z "${last_tag}" ]]; then
+		last_tag=$(git tag --sort=version:refname | tail -n 1)
+	fi
+fi
+
+version="${last_tag}"
+
+# If the HEAD has extra commits since the last tag then we are in a dev version.
+#
+# Dev versions are denoted by the "-devel+" suffix with a trailing commit short
```
Comment (mafredri) [inline scripts/version.sh:51]: We shouldn't inherit `-devel+` from coder/coder, it's not totally semver compatible. Preferable would be `+devel-commit` or something like that.

File: scripts/version.sh [hunk 4]
```diff
@@ -1,10 +1,77 @@
...
+else
+	version+="-dev-$(git rev-parse --short HEAD)"
+fi
 
-last_tag="$(git describe --tags --abbrev=0)"
-version="$last_tag"
+# If the git repo has uncommitted changes, mark the version string as 'dirty'.
+dirty_files=$(git ls-files --other --modified --exclude-standard)
+if [[ -n "${dirty_files}" ]]; then
+	version+="-dirty"
```
Comment (mafredri) [inline scripts/version.sh:73]:
```suggestion
	version+="+dirty"
```

`-` is reserved for release type, e.g. `alpha`. `+` is for build information.

https://semver.org

**Review by mafredri — APPROVED** [review]
We should probably strip out more cruft that we don't want to inherit, but no major blockers for now so approving.

File: scripts/version.sh (hunk 1 above)
Comment (johnstcn) [reply scripts/version.sh:30]: nope, copypasta

File: scripts/version.sh (hunk 2 above)
Comment (johnstcn) [reply scripts/version.sh:35]: v1.0.0, wdyt?

File: scripts/version.sh (hunk 3 above)
Comment (johnstcn) [reply scripts/version.sh:51]: It seems like a valid pre-release identifier to me?

**Pull request merged**